### PR TITLE
Add CBC to STPCardFormView

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
* Refactors CBC support from STPPaymentCardTextField into STPCBCController.
* STPCardFormView now has a CBC controller, which it uses to override the card brand for CBC.
* Removed `STPPaymentMethodCardParams.brand(for:)`, which was an incorrect copy of `STPCard.brand(for:)`. Not sure why we had this!
* Removed some now-unused code to embed STPCardFormView in PaymentSheet.

## Motivation
Adding CBC to STPCardFormView.

## Testing
Added new snapshot tests and unit tests for CBC scenarios.

## Changelog
Not adding one until all CBC work is in.

### CODE REVIEW

**Comment by davidme-stripe** [comment]
The snapshot changes away from the PaymentSheet style are intentional — as far as I can tell, we only used that style in PaymentSheet, and we no longer do.

**Review by porter-stripe — DISMISSED** [review]
Nothing blocking! Just a few questions/comments

File: Stripe/StripeiOSTests/STPPaymentCardTextFieldViewModelTest.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ class STPPaymentCardTextFieldViewModelTest: XCTestCase {
 
     override func setUp() {
         super.setUp()
-        viewModel = STPPaymentCardTextFieldViewModel()
+        viewModel = STPPaymentCardTextFieldViewModel(brandUpdateHandler: {})
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/StripeiOSTests/STPPaymentCardTextFieldViewModelTest.swift:22]: I wanted to make sure it was never left as nil in the code, as there's no situation (outside of this specific test) where it should be nil.

File: Stripe/StripeiOS/Source/STPCardScanner.swift [hunk 2]
```diff
@@ -30,7 +30,7 @@ enum STPCardScannerError: Int {
 
 @available(macCatalyst 14.0, *)
 @objc(STPCardScanner_legacy)
-class STPCardScanner: NSObject, AVCaptureVideoDataOutputSampleBufferDelegate, STPCardScanningProtocol {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/StripeiOS/Source/STPCardScanner.swift:33]: STPCardScanningProtocol is now unused, as STPCardFormView no longer uses it. (I think this was triggering a warning.)

File: Stripe/StripeiOSTests/STPPaymentMethodCardTest.swift [hunk 3]
```diff
@@ -91,35 +91,35 @@ class STPPaymentMethodCardTest: XCTestCase {
     }
 
     func testBrandFromString() {
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/StripeiOSTests/STPPaymentMethodCardTest.swift:93]: Yep

[the remark this answers is not in the document]
Comment (davidme-stripe) [reply Stripe/StripeiOSTests/STPPaymentMethodCardTest.swift]: Done!

**Pull request merged**